tests/Core/Files/FileList/AddFileTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Tests for the \PHP_CodeSniffer\Files\FileList::addFile method.
+ *
+ * @copyright 2019-2025 PHPCSStandards Contributors
+ * @license   https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Files\FileList;
+
+use PHP_CodeSniffer\Files\FileList;
+use PHP_CodeSniffer\Tests\ConfigDouble;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+/**
+ * Tests for the \PHP_CodeSniffer\Files\FileList::addFile method.
+ *
+ * @covers \PHP_CodeSniffer\Files\FileList::addFile
+ */
+final class AddFileTest extends TestCase
+{
+
+    /**
+     * The FileList object.
+     *
+     * @var \PHP_CodeSniffer\Files\FileList
+     */
+    private $fileList;
+
+
+    /**
+     * Initialize the FileList object.
+     *
+     * @before
+     *
+     * @return void
+     */
+    protected function initializeFileList()
+    {
+        $config         = new ConfigDouble();
+        $config->filter = __DIR__.'/FilterDouble.php';
+        $config->files  = [];
+        $ruleset        = $this->getMockBuilder('PHP_CodeSniffer\Ruleset')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $this->fileList = new FileList($config, $ruleset);
+
+    }//end initializeFileList()
+
+
+    /**
+     * Test adding a file to the list.
+     *
+     * @param string      $fileName   The name of the file to add.
+     * @param object|null $fileObject An optional file object to add instead of creating a new one.
+     *
+     * @dataProvider dataAddFile
+     *
+     * @return void
+     */
+    public function testAddFile($fileName, $fileObject=null)
+    {
+        $this->assertCount(0, $this->fileList);
+
+        $this->fileList->addFile($fileName, $fileObject);
+
+        $fileListArray = iterator_to_array($this->fileList);
+
+        $this->assertCount(1, $this->fileList, 'File count mismatch');
+        $this->assertArrayHasKey($fileName, $fileListArray, 'File not found in list');
+
+        if (isset($fileObject) === true) {
+            $this->assertSame($fileObject, $fileListArray[$fileName], 'File object mismatch');
+        } else {
+            $this->assertInstanceOf(
+                'PHP_CodeSniffer\Files\File',
+                $fileListArray[$fileName],
+                'File object not found in list'
+            );
+        }
+
+    }//end testAddFile()
+
+
+    /**
+     * Data provider for testAddFile.
+     *
+     * @return array<string, array<string>>
+     */
+    public static function dataAddFile()
+    {
+        return [
+            'Regular file'                  => ['fileName' => 'test1.php'],
+            'STDIN'                         => ['fileName' => 'STDIN'],
+            'Regular file with file object' => [
+                'fileName'   => 'test1.php',
+                'fileObject' => new stdClass(),
```

Comment:
This should create a `File` object, not a `stdClass` object. As things are, this will break in PHPCS 4.0 when the type declarations are added (`addFile(string $path, ?File $file=null)`).

Reply:
Good catch! I added a commit creating an instance of the `File` class. To do that, I had to instantiate both `ConfigDouble` and `Ruleset`. Let me know if you had a different approach in mind.